[CCI] Create index template guide (#148)
opensearch-project/opensearch-rs: guides/index_template.md

# Index Template

Index templates are a convenient way to define settings, mappings, and aliases for one or more indices when they are created. In this guide, you'll learn how to create an index template and apply it to an index.

## Setup

Assuming you have OpenSearch running locally on port 9200, you can create a client instance with the following code:

```rust
// Create a client to make API calls to OpenSearch running on http://localhost:9200.

> let client = OpenSearch::default();

Reviewer: Sorry missed that before - this line is not needed since the client is created few lines below (line #15)

// Alternatively, you can create a client to make API calls against OpenSearch running on a specific url::Url.

> let url = Url::parse("https://example.com")?;

Reviewer: Please 

```suggestion
let url = Url::parse("https://localhost:9200")?; 
```

The `example.com` is a real site, the port is required

Author: Yes, sorry, corrected it!

Reviewer: Hm .. I still see it as it was before ...

# Index Template

Index templates are a convenient way to define settings, mappings, and aliases for one or more indices when they are created. In this guide, you'll learn how to create an index template and apply it to an index.

## Setup

Assuming you have OpenSearch running locally on port 9200, you can create a client instance

> with the following code:

Reviewer 2: move `with` up the previous line, don't line break 

https://code.dblock.org/2021/06/07/to-wrap-or-not-to-wrap-in-markdown.html

# Index Template

Index templates are a convenient way to define settings, mappings, and aliases for one or more indices when they are created. In this guide, you'll learn how to create an index template and apply it to an index.

## Setup

Assuming you have OpenSearch running locally on port 9200, you can create a client instance with the following code:

```rust

> // Let's create a client instance, and an index named `movies`:
> 
> // Create a client to make API calls to OpenSearch running on http://localhost:9200.
> 
> let url = Url::parse("https://localhost:9200")?;
> let transport = TransportBuilder::new(SingleNodeConnectionPool::new(url))
>     .cert_validation(CertificateValidation::None)
>     .build()?;
> let client = OpenSearch::new(transport);

Reviewer 3: Let's stay consistent with #146 and have the same setup as that, also the first comment refers to creating an index named `movies` which we are not doing, so remove it.
```suggestion
// Create a client to make API calls to OpenSearch running on https://localhost:9200.

let url = Url::parse("https://localhost:9200")?;
let credentials = Credentials::Basic("admin".into(), "admin".into());
let transport = TransportBuilder::new(SingleNodeConnectionPool::new(url))
      .cert_validation(CertificateValidation::None)
      .auth(credentials)
      .build()?;
let client = OpenSearch::new(transport);
```

```rust
client.cluster().put_component_template(ClusterPutComponentTemplateParts::Name("books_mappings"))
    .body(json!({
        "template": {
            "mappings": {
                "properties": {
                    "title": { "type": "text" },
                    "author": { "type": "text" },
                    "published_on": { "type": "date" },
                    "pages": { "type": "integer" }
                }
            }
        }
    }))
    .send()
    .await?;

client.indices().put_index_template(IndicesPutIndexTemplateParts::Name("books"))
    .body(json!({
        "index_patterns": ["books-*"],
        "composed_of": ["books_mappings"], // use the `books_mappings` component template
        "priority": 0,
        "template": {
            "settings": {
                "index": {
                    "number_of_shards": 3,
                    "number_of_replicas": 0
                }
            }
        }
    }))
    .send()
    .await?;

> client.indices().put_index_template(IndicesPutIndexTemplateParts::Name("books"))
>     .body(json!({
>         "index_patterns": ["books-*"],

Reviewer 3: This should be the `books-fiction` template
```suggestion
client.indices().put_index_template(IndicesPutIndexTemplateParts::Name("books-fiction"))
    .body(json!({
        "index_patterns": ["books-fiction-*"],
```
